src/core/mesh/qgstopologicalmesh.h

```diff
@@ -0,0 +1,277 @@
+/***************************************************************************
+  qgstopologicalmesh.h - QgsTopologicalMesh
+
+ ---------------------
+ begin                : 18.6.2021
+ copyright            : (C) 2021 by Vincent Cloarec
+ email                : vcloarec at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSTOPOLOGICALMESH_H
+#define QGSTOPOLOGICALMESH_H
+
+#include "qgsmeshdataprovider.h"
+
+SIP_NO_FILE
+
+class QgsMeshEditingError;
+class QgsMeshVertexCirculator;
+
+/**
+ * \ingroup core
+ *
+ * \brief Class that wraps a QgsMesh to ensure the consistency of the mesh during editing and help to access to elements from other elements
```

Comment:
what is "element" face/edge?

Reply:
here elements are vertex/faces (edge not yet supported), the class allow  vertices <--> faces access in both direction.